backend/test_optimization.py

```
@@ -0,0 +1,111 @@
+import json
+
+BASE_URL = "http://localhost:5000"
+
+def test_endpoint(endpoint, payload, minimal=False):
+    """Test an endpoint and show response size."""
+    url = f"{BASE_URL}{endpoint}"
+    if minimal:
+        url += "?minimal=true"
+    
+    try:
+        response = requests.post(url, json=payload, headers={
```

Ruff (0.15.7)

[error] Probable use of `requests` call without timeout

(S113)